### PR TITLE
Fix #135. Add Explicit options for adding ACF Field Groups to the Schema

### LINKED ISSUE #135: Add Explicit options for adding ACF Field Groups to the Schema
Currently, the way WPGraphQL for ACF implicitly maps field groups from ACF Location Rules to the WPGraphQL Schema causes a lot of confusion. 

ACF has a lot of location rules that require specific run-time context that's not available for Schema generation. 

For example, rules such as `Page is not equal to Home` or `Post is equal to Hello World` don't exactly map well to a Schema. 

The Schema is representative of all Types available, and context such as whether a Page is the Home Page, or a Post is the "Hello World" post is runtime context. 

So, in an effort to reduce this confusion, I'm working on a more explicit setting where you can define the GraphQL Types to show the ACF Field Group on, and _not_ rely on the location rules at all. 

The location rules can be used for determining the editing experience, and the new setting can be used for determining explicitly which Types the field group should be available on in the Schema. 

Something like the following: 

Screen Shot 2020-04-20 at 1 56 00 PM

This checkbox field will show all Types that _can_ have field groups associated with them. Post Types and Taxonomies that are exposed to GraphQL, Users, Comments, Menus, Menu Items, Media, Page Templates, etc.

### PR DESCRIPTION
**Pull request by rsm0128**
1. Added Explicit options for adding ACF Field Groups to the Schema
2. Added explicit option auto-generation code for plugin upgrade from older versions.

### CODE REVIEW

**Comment by jasonbahl** [comment]
@rsm0128 This is a great PR! I'll review it in detail shortly!

Thanks for working on this 🙏

File: src/class-acfsettings.php [hunk 1]
```diff
@@ -109,6 +109,19 @@ public function add_field_group_settings( $field_group ) {
 			]
 		);
 
+		$choices = Config::get_all_graphql_types();
+		acf_render_field_wrap(
+			[
+				'label'        => __( 'GraphQL Types to Show the Field Group On', 'wp-graphql-acf' ),
+				'instructions' => __( 'Select the Types in the WPGraphQl Schema to show the fields in this fiedl group on', 'wp-graphql-acf' ),
```
Comment (drewbaker) [inline src/class-acfsettings.php]: Spelling error. Should be `field`.

**Comment by drewbaker** [comment]
Excited to see this working!

File: src/class-acfsettings.php (hunk 1 above)
Comment (rsm0128) [reply src/class-acfsettings.php]: Fixed the spelling. Thanks

**Comment by Xedecimal** [comment]
I checked this branch out and tested it. With this you could pick something like Category (taxonomy), but you couldn't pick a specific category for it to show on. Also in WooCommerce, productCategory has the same name as regular post categories so they both show up here as Category (taxonomy).

I don't think this solves the issue of being able to iterate over all available field groups for a given type. I'm thinking you have a store and products in a given productCategory have a series of field groups that could be applied to products inside a given category, or to all categories store-wide. We don't want to hunt for a set of specific names every time we load a product category but to simply get all of the field groups that apply to this context (viewing a category, or viewing any product), then we could show all of them.

This does solve the problem of specifying multiple Location conditions not being shown at all ever. I could apply a complex ACF Field Group to all categories and they would show up everywhere with this.

**Comment by drewbaker** [comment]
So the idea is that you opt into what TYPEs you want the field group to be on (as that is how the GQL schema works) and then you'd use the ACF location filters (and [their existing API](https://www.advancedcustomfields.com/resources/custom-location-rules-v5-8/)) for dictating where the field groups show up. This should allow you to say "I want this field group on CATEGORIES but only show up on the category called CARS). Does that help?

**Comment by Xedecimal** [comment]
I think my request is outside the scope of this fix. This does allow items to show up when you specify multiple locations. My request is more along the lines of getting an array of all the field groups and properties that apply to a given item. Which is currently not possible with the layout of wp-graphql-acf in general. You would need to know each name of each property you are asking for in graphql, of which I would need a more generic list of edges and nodes for all groups and all fields that apply to an item like a category. I've started developing a resolver to get edges for all fields but it's getting more and more complex. Namely there's no good generic way to specify all different types of fields as of yet. I would say this PR is good and works for what it's meant to.

**Comment by intelligence** [comment]
Thanks for this, with my limited testing, it works well!

**Comment by drewbaker** [comment]
@rsm0128 I spoke with @jasonbahl and he'd like to get some test coverage for this PR. Can you do that (obviously I'm happy to pay for it). Any questions on setting up the tests, ask here. Thanks!

**Comment by rsm0128** [comment]
Thanks @drewbaker !
@jasonbahl Could you guide me on setting up the tests? @drewbaker is eager to see this PR be merged soon!
So I'm willing to assist you on this.

**Comment by rsm0128** [comment]
@jasonbahl @drewbaker 
By mistake, I created this PR from master branch of my fork, and it made it hard to check and update new commits.
So I created another [PR](https://github.com/wp-graphql/wp-graphql-acf/pull/229) from my fix branch.

**Pull request closed without merging**